src/time.cr

```diff
@@ -925,6 +925,36 @@ struct Time
     time - time.offset.seconds
   end
 
+  # Creates an instance representing the *week*-th *day_of_week* in the given
+  # *year* and *month*.
+  #
+  # If *week* is `5` and there are not enough weeks in the given month (e.g.
+  # February in a non-leap year), the last week of the month is used instead.
+  #
+  # Valid value ranges for the individual fields:
+  #
+  # * `year`: `1..9999`
+  # * `month`: `1..12`
+  # * `week`: `1..5`
+  # * `day_of_week`: `0..7`
```

Comment:
I think this should be either `1..7` or `0..6` (or `0...7`)

Reply:
This is congruent with `DayOfWeek.from_value` which accepts either 0 or 7 for Sunday.

Reply:
In fact, I think there could be some value in using `DayOfWeek.from_value` here. It implicitly handles domain errors.
Perhaps the parameter restriction should even be narrowed to `DayOfWeek`, instead of the union?
An enum value is more explicit than a plain number.

I realize that `.week_date` has union type restriction as well, so that would be a change.